core_lib/src/structure/filemanager.cpp

```diff
@@ -64,7 +69,15 @@ Object* FileManager::load(const QString& sFileName)
     {
         dd << "Recognized New zipped Pencil2D File Format (*.pclx) !";
 
-        unzip(sFileName, obj->workingDir());
+        Status sanityCheck = MiniZ::sanityCheck(sFileName);
+
+        // Let's check if we can read the file before we try to unzip.
+        if (!sanityCheck.ok()) {
+            dd.collect(sanityCheck.details());
+        } else {
+            Status unzipStatus = unzip(sFileName, obj->workingDir());
+            dd.collect(unzipStatus.details());
+        }
```

Comment:
We don't do much beyond that point, that would cause further problems I think, so I left it there because we would get more data for the report. If it fails to unzip, then it should be caught by one of the following checks.